### PR TITLE
feat(*): Adds host_core building to CI pipeline

### PR DESCRIPTION
**Pull request by thomastaylor312**
The release pipeline is now its own separate file and builds tarballs for
both the full host and the core only.

You can see a test run of this working [here](https://github.com/thomastaylor312/wasmcloud-otp/runs/4304561726?check_suite_focus=true)

### CODE REVIEW

**Comment by autodidaddict** [comment]
@thomastaylor312  build failure:
```
Error: 7.347 [error] Unable to load crypto library. Failed with error:
":load_failed, Failed to load NIF library /home/runner/work/_temp/.setup-elixir/otp/lib/crypto-4.6.5/priv/lib/crypto: 'libcrypto.so.1.0.0: cannot open shared object file: No such file or directory'"
OpenSSL might not be installed on this system.
```

**Comment by thomastaylor312** [comment]
Looks like it was an issue with me updating the ubuntu version. Should be fixed

**Review by brooksmtownsend — DISMISSED** [review]
A couple small typos, ~only other request is to consider forking wasmcloud-otp, disabling the "github release" and docker release parts and just ensure that the tarballs build well and attach to the action in the fork.~

Edit again: that's what I get for not reading the description better. Looks like there shows the only requests that I have, it skips the npm steps for both wasmcloud_host and host_core ATM

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+      # Setup Rust for NIF dependencies
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          target: ${{ matrix.rust-target }}
+          override: true
+
+      # Setup node for phoenix deps
+      - uses: actions/setup-node@v2
+        if: ${{ env.working-director == 'wasmcloud_host' }}
```
[the remark this answers is not in the document]
Comment (thomastaylor312) [reply .github/workflows/release.yml:118]: 🤦

**Review by autodidaddict — APPROVED** [review]

**Pull request merged**